src/main/java/com/ccs/inventorymanagement/route/CreateClothingHandler.java

```diff
@@ -0,0 +1,94 @@
+package com.ccs.inventorymanagement.route;
+
+import com.ccs.inventorymanagement.domain.Clothing;
+import com.ccs.inventorymanagement.domain.Item;
+import com.ccs.inventorymanagement.service.ClothingService;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.springframework.web.reactive.function.BodyInserters;
+import org.springframework.web.reactive.function.server.HandlerFunction;
+import org.springframework.web.reactive.function.server.ServerRequest;
+import org.springframework.web.reactive.function.server.ServerResponse;
+import reactor.core.publisher.Mono;
+
+import java.time.Instant;
+import java.util.UUID;
+
+public class CreateClothingHandler implements HandlerFunction<ServerResponse> {
+
+    private ClothingService clothingService;
+
+    public CreateClothingHandler(ClothingService clothingService) {
+        this.clothingService = clothingService;
+    }
+    @Override
+    public Mono<ServerResponse> handle(ServerRequest serverRequest) {
+        return serverRequest.bodyToMono(Request.class)
+                .flatMap(request -> clothingService.create(Clothing.builder()
+                        .name(request.getName())
+                        .condition(request.getCondition())
+                        .status(request.getStatus())
+                        .description(request.getDescription())
+                        .brand(request.getBrand())
+                        .color(request.getColor())
+                        .type(request.getType())
+                        .gender(request.getGender())
+                        .size(request.getSize())
+                        .build()))
+                .flatMap(clothing -> ServerResponse.ok()
+                        .body(BodyInserters.fromValue(Response.from(clothing))));
```

Comment:
What happens if you get an empty or an error? What Response do you want. It seems to be missing here

Reply:
Fixed, please review.